relay-server/src/actors/upstream.rs

```diff
@@ -1010,6 +1065,10 @@ enum Action {
     ///
     /// The entry is placed on the front of the [`UpstreamQueue`].
     Retry(Entry),
+    /// Notifies a request has been completed successfully to upstream.
+    ///
+    /// Successful requests reset the retry backoff.
+    SuccessfulRequest,
```

Comment:
This action is redundant. There's already a `Complete` with a request outcome that could reset the backoff.

Reply:
the complete message does contain whether it was succeeded or not though, so I think you could do it in `fn complete`